File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/data/network/api/LockApi.java
package com.lattis.ellipse.data.network.api;

import com.lattis.ellipse.data.network.model.body.lock.SignedMessagePublicKeyBody;
import com.lattis.ellipse.data.network.model.response.lock.SignedMessagePublicKeyResponse;

import retrofit2.http.Body;
import retrofit2.http.POST;
import io.reactivex.Observable;

public interface LockApi {


    @POST("locks/signed-message-and-public-key-for-trip")
    Observable<SignedMessagePublicKeyResponse> getSignedMessagePublicKey(@Body SignedMessagePublicKeyBody signedMessagePublicKeyBody);


}

File: velolabs_repos/lattis_android_v2/data/src/main/java/com/lattis/data/repository/datasources/api/lock/LockApi.kt
package com.lattis.data.repository.datasources.api.lock

import com.lattis.data.entity.body.lock.SignedMessagePublicKeyBody
import com.lattis.data.entity.response.lock.SignedMessagePublicKeyResponse
import com.lattis.data.entity.response.lock.TapkeyAccessResponse
import io.reactivex.rxjava3.core.Observable
import retrofit2.http.*

interface LockApi {
    @POST("api/locks/signed-message-and-public-key-for-trip")
    fun getSignedMessagePublicKey(@Body signedMessagePublicKeyBody: SignedMessagePublicKeyBody): Observable<SignedMessagePublicKeyResponse>


    @GET("api/tapkey/credentials/{mac_id}")
    fun getTapkeyAccess(@Path("mac_id") mac_id:String,@Query("fleetId") fleetId:Int): Observable<TapkeyAccessResponse>
}